### ISSUE TITLE
Failed to install Microsoft Edge when $home is located on ext. drive

### ISSUE THREAD

**Issue by okleinschmidt** [Homebrew/homebrew-cask#177636]
### Verification

- [X] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).
- [X] I have retried my command with `--force`.
- [x] I ran `brew update-reset && brew update` and retried my command.
- [X] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [X] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [X] I made doubly sure this is not a [checksum does not match / SHA256 mismatch](https://docs.brew.sh/Common-Issues#cask---checksum-does-not-match) error (do not open an issue before trying to open a PR to fix first).

### Description of issue

The latest version of Microsoft Edge can no longer be installed. The root cause is apparently that my $home is on an external NVMe drive. The manual installation, without brew, works without any problems. The same problem occurs with Microsoft Powershell.

Of course, swapping the home directory to an external drive is not an everyday occurrence. But maybe this problem is simple and quick to fix.

### Command that failed

brew install --cask microsoft-edge

### Output of command with `--verbose --debug`

```shell
==> Downloading https://formulae.brew.sh/api/cask.jws.json
/usr/bin/env /opt/homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --user-agent Homebrew/4.3.6\ \(Macintosh\;\ arm64\ Mac\ OS\ X\ 14.5\)\ curl/8.6.0 --header Accept-Language:\ en --fail --remote-time --output /Volumes/Datastore/USERNAME/Library/Caches/Homebrew/api/cask.jws.json --location --time-cond /Volumes/Datastore/USERNAME/Library/Caches/Homebrew/api/cask.jws.json --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/4.3.6\ \(Macintosh\;\ arm64\ Mac\ OS\ X\ 14.5\)\ curl/8.6.0 --header Accept-Language:\ en --fail --compressed --speed-limit 100 --speed-time 5 https://formulae.brew.sh/api/cask.jws.json
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1444k  100 1444k    0     0  6014k      0 --:--:-- --:--:-- --:--:-- 5993k
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading microsoft-edge
==> Cask::Installer#install
==> Printing caveats
==> Cask::Installer#fetch
==> Downloading https://msedge.sf.dl.delivery.mp.microsoft.com/filestreamingservice/files/24e2a4e2-9bc4-4756-8085-3b4bbc280928/MicrosoftEdge-126.0.2592.68.pkg
/usr/bin/env /opt/homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/4.3.6\ \(Macintosh\;\ arm64\ Mac\ OS\ X\ 14.5\)\ curl/8.6.0 --header Accept-Language:\ en --retry 3 --fail --location --silent --head https://msedge.sf.dl.delivery.mp.microsoft.com/filestreamingservice/files/24e2a4e2-9bc4-4756-8085-3b4bbc280928/MicrosoftEdge-126.0.2592.68.pkg
/usr/bin/env /opt/homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/4.3.6\ \(Macintosh\;\ arm64\ Mac\ OS\ X\ 14.5\)\ curl/8.6.0 --header Accept-Language:\ en --retry 3 --fail --location --silent --head --request GET https://msedge.sf.dl.delivery.mp.microsoft.com/filestreamingservice/files/24e2a4e2-9bc4-4756-8085-3b4bbc280928/MicrosoftEdge-126.0.2592.68.pkg
Already downloaded: /Volumes/Datastore/USERNAME/Library/Caches/Homebrew/downloads/027f39ebf385018d706185caecc9c99c2e448b266a4657d45b125babd5405f98--MicrosoftEdge-126.0.2592.68.pkg
==> Checking quarantine support
/usr/bin/env /usr/bin/xattr -h
/usr/bin/env /Library/Developer/CommandLineTools/usr/bin/swift -target arm64-apple-macosx14 /opt/homebrew/Library/Homebrew/cask/utils/quarantine.swift
==> Quarantine is available.
==> Verifying Gatekeeper status of /Volumes/Datastore/USERNAME/Library/Caches/Homebrew/downloads/027f39ebf385018d706185caecc9c99c2e448b266a4657d45b125babd5405f98--MicrosoftEdge-126.0.2592.68.pkg
/usr/bin/env /usr/bin/xattr -p com.apple.quarantine /Volumes/Datastore/USERNAME/Library/Caches/Homebrew/downloads/027f39ebf385018d706185caecc9c99c2e448b266a4657d45b125babd5405f98--MicrosoftEdge-126.0.2592.68.pkg
==> /Volumes/Datastore/USERNAME/Library/Caches/Homebrew/downloads/027f39ebf385018d706185caecc9c99c2e448b266a4657d45b125babd5405f98--MicrosoftEdge-126.0.2592.68.pkg is quarantined
==> Verifying checksum for '027f39ebf385018d706185caecc9c99c2e448b266a4657d45b125babd5405f98--MicrosoftEdge-126.0.2592.68.pkg'
/usr/bin/env hdiutil imageinfo -format /Volumes/Datastore/USERNAME/Library/Caches/Homebrew/downloads/027f39ebf385018d706185caecc9c99c2e448b266a4657d45b125babd5405f98--MicrosoftEdge-126.0.2592.68.pkg
==> Installing Cask microsoft-edge
==> Cask::Installer#stage
==> Extracting primary container
==> Using container class UnpackStrategy::Pkg for /Volumes/Datastore/USERNAME/Library/Caches/Homebrew/downloads/027f39ebf385018d706185caecc9c99c2e448b266a4657d45b125babd5405f98--MicrosoftEdge-126.0.2592.68.pkg
/usr/bin/env cp -p -c /Volumes/Datastore/USERNAME/Library/Caches/Homebrew/downloads/027f39ebf385018d706185caecc9c99c2e448b266a4657d45b125babd5405f98--MicrosoftEdge-126.0.2592.68.pkg /opt/homebrew/Caskroom/microsoft-edge/126.0.2592.68,24e2a4e2-9bc4-4756-8085-3b4bbc280928/MicrosoftEdge-126.0.2592.68.pkg
cp: /opt/homebrew/Caskroom/microsoft-edge/126.0.2592.68,24e2a4e2-9bc4-4756-8085-3b4bbc280928/MicrosoftEdge-126.0.2592.68.pkg: clonefile failed: Cross-device link
==> Purging files for version 126.0.2592.68,24e2a4e2-9bc4-4756-8085-3b4bbc280928 of Cask microsoft-edge
Error: Failure while executing; `/usr/bin/env cp -p -c /Volumes/Datastore/USERNAME/Library/Caches/Homebrew/downloads/027f39ebf385018d706185caecc9c99c2e448b266a4657d45b125babd5405f98--MicrosoftEdge-126.0.2592.68.pkg /opt/homebrew/Caskroom/microsoft-edge/126.0.2592.68,24e2a4e2-9bc4-4756-8085-3b4bbc280928/MicrosoftEdge-126.0.2592.68.pkg` exited with 1. Here's the output:
cp: /opt/homebrew/Caskroom/microsoft-edge/126.0.2592.68,24e2a4e2-9bc4-4756-8085-3b4bbc280928/MicrosoftEdge-126.0.2592.68.pkg: clonefile failed: Cross-device link

/opt/homebrew/Library/Homebrew/system_command.rb:352:in `assert_success!'
/opt/homebrew/Library/Homebrew/system_command.rb:75:in `run!'
/opt/homebrew/Library/Homebrew/system_command.rb:40:in `run'
/opt/homebrew/Library/Homebrew/system_command.rb:44:in `run!'
/opt/homebrew/Library/Homebrew/utils/copy.rb:24:in `with_attributes'
/opt/homebrew/Library/Homebrew/unpack_strategy/uncompressed.rb:27:in `extract_to_dir'
/opt/homebrew/Library/Homebrew/unpack_strategy.rb:114:in `extract'
/opt/homebrew/Library/Homebrew/unpack_strategy/uncompressed.rb:20:in `extract_nestedly'
/opt/homebrew/Library/Homebrew/cask/installer.rb:225:in `extract_primary_container'
/opt/homebrew/Library/Homebrew/cask/installer.rb:82:in `stage'
/opt/homebrew/Library/Homebrew/cask/installer.rb:109:in `install'
/opt/homebrew/Library/Homebrew/cmd/install.rb:246:in `block in run'
/opt/homebrew/Library/Homebrew/cmd/install.rb:235:in `each'
/opt/homebrew/Library/Homebrew/cmd/install.rb:235:in `run'
/opt/homebrew/Library/Homebrew/brew.rb:92:in `<main>'
```

### Output of `brew doctor` and `brew config`

```shell
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

$ brew config
Unexpected dylibs:
  /usr/local/lib/Spyder.dylib
  /usr/local/lib/Spyder3.dylib
  /usr/local/lib/Spyder5.dylib
  /usr/local/lib/dccmtr.dylib
  /usr/local/lib/dcxmtr.dylib
  /usr/local/lib/libCIEHandler.dylib
  /usr/local/lib/libCalibrationHandler.dylib
  /usr/local/lib/libDeviceManager.dylib
  /usr/local/lib/libErrorHandler.dylib
  /usr/local/lib/libICMManager.dylib
  /usr/local/lib/libMCCAppManager.dylib
  /usr/local/lib/libMCCEngineWrapper.dylib
  /usr/local/lib/libMonitorDDCCISDK.dylib
  /usr/local/lib/libProtocolEngine.dylib
  /usr/local/lib/libmkusb.dylib
  /usr/local/lib/libusb-1.0.0.dylib

HOMEBREW_VERSION: 4.3.6
ORIGIN: https://github.com/Homebrew/brew
HEAD: e8430b25a1d1321f32e5093d62b57b5cb7cfb3c3
Last commit: 8 days ago
Core tap JSON: 24 Jun 05:04 UTC
Core cask tap JSON: 24 Jun 05:14 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: /usr/bin/vim
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.3.3 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.3/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew tap`

```shell
$ brew tap
derailed/k9s
homebrew/bundle
kubecolor/tap
natesales/repo
teamookla/speedtest
vladimirvivien/oss-tools
```

**Comment by khipp** [comment]
Does this problem persist on the newly released Homebrew v4.3.7? There has been a fix regarding file copying.

**Comment by okleinschmidt** [comment]
Thanks, 4.3.7 works again!
